central/delegatedregistryconfig/service/service.go

```diff
@@ -0,0 +1,213 @@
+
+	AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error)
+
+	v1.DelegatedRegistryConfigServiceServer
+}
+
+// New returns a new Service instance using the given DataStore.
+func New(dataStore datastore.DataStore, clusterDataStore cluster.DataStore) Service {
+	return &serviceImpl{
+		dataStore:        dataStore,
+		clusterDataStore: clusterDataStore,
+	}
+}
+
+type serviceImpl struct {
+	v1.UnimplementedDelegatedRegistryConfigServiceServer
+
+	dataStore        datastore.DataStore
+	clusterDataStore cluster.DataStore
+}
+
+// RegisterServiceServer registers this service with the given gRPC Server.
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	v1.RegisterDelegatedRegistryConfigServiceServer(grpcServer, s)
+}
+
+// RegisterServiceHandler registers this service with the given gRPC Gateway endpoint.
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	return v1.RegisterDelegatedRegistryConfigServiceHandler(ctx, mux, conn)
+}
+
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, authorizer.Authorized(ctx, fullMethodName)
+}
+
+// GetConfig returns Central's delegated registry config
+func (s *serviceImpl) GetConfig(ctx context.Context, _ *v1.Empty) (*v1.DelegatedRegistryConfig, error) {
+	config, exists, err := s.dataStore.GetConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving config %w", err)
+	}
+
+	if !exists {
+		return &v1.DelegatedRegistryConfig{}, nil
+	}
+
+	return convert.StorageToAPI(config), nil
+}
+
+// GetClusters returns the list of clusters (id + name) that are eligible for delegating scanning
+// requests (ie: only clusters with scanners that understand the delegated registry config)
+func (s *serviceImpl) GetClusters(ctx context.Context, _ *v1.Empty) (*v1.DelegatedRegistryClustersResponse, error) {
+	clusters, err := s.getClusters(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving clusters %w", err)
+	}
+
+	if len(clusters) == 0 {
+		return nil, status.Error(codes.NotFound, "no valid clusters found")
+	}
+
+	return &v1.DelegatedRegistryClustersResponse{
+		Clusters: clusters,
+	}, nil
+}
+
+// PutConfig updates Central's delegated registry config
+func (s *serviceImpl) PutConfig(ctx context.Context, config *v1.DelegatedRegistryConfig) (*v1.DelegatedRegistryConfig, error) {
+	if err := s.validate(ctx, config); err != nil {
+		return nil, fmt.Errorf("%w: %v", errox.InvalidArgs, err.Error())
+	}
+
+	if err := s.dataStore.UpsertConfig(ctx, convert.APIToStorage(config)); err != nil {
+		return nil, fmt.Errorf("upserting config %w", err)
+	}
+
+	return config, nil
+}
+
+func (s *serviceImpl) validate(ctx context.Context, config *v1.DelegatedRegistryConfig) error {
+	if config == nil {
+		// this block not reachable via GRPC-gateway invocations
```

Comment:
is it part of the path?

Reply:
the only current known consumer of this endpoint will be the UI via the grpc-gateway, so not in that path in that regards, I believe it's possible a GRPC consumer could reach that path.  I can remove that comment if its misleading